### ISSUE TITLE
Add support for pydantic > 2

### ISSUE THREAD

**Issue by lyz-code** [lincolnloop/goodconf#40]
Hi! when loading `GoodConf` I'm getting this deprecation warning error:

```
src/test.py:7: in <module>
    from goodconf import GoodConf
../../.local/share/virtualenvs/test/lib/python3.9/site-packages/goodconf/__init__.py:12: in <module>
    from pydantic import BaseSettings, PrivateAttr
../../.local/share/virtualenvs/test/lib/python3.9/site-packages/pydantic/__init__.py:374: in __getattr__
    return _getattr_migration(attr_name)
../../.local/share/virtualenvs/test/lib/python3.9/site-packages/pydantic/_migration.py:296: in wrapper
    raise PydanticImportError(
E   pydantic.errors.PydanticImportError: `BaseSettings` has been moved to the `pydantic-settings` package. See https://docs.pydantic.dev/2.6/migration/#basesettings-has-moved-to-pydantic-set
tings for more details.
E   
E   For further information visit https://errors.pydantic.dev/2.6/u/import-error
```

I just wanted to let you know.

**Comment by lyz-code** [comment]
I started working on it but it's been more difficult than I thought. I leave you [the progress I've made](https://github.com/lincolnloop/goodconf/pull/41) but I won't be able to work more on it anymore in a close future :(

**Comment by ipmb** [comment]
Thanks @lyz-code! That was my experience when I picked it up as well. I have a PR that is also WIP here: https://github.com/lincolnloop/goodconf/pull/36

I'll review and see if we can combine forces :)

**Comment by ipmb** [comment]
I found out that there is a copy of the V1 features available in Pydantic v2. This may be a good temporary stop-gap:

https://docs.pydantic.dev/latest/migration/#continue-using-pydantic-v1-features

> The Pydantic V2 package also continues to provide access to the Pydantic V1 API by importing through `pydantic.v1`.

**Comment by dchukhin** [comment]
This was completed in #36, and released in [v5.0.0](https://pypi.org/project/goodconf/5.0.0/)
